### PR TITLE
tests: nested_irq: Fix k_busy_wait usage and interrupt priority

### PR DESCRIPTION
**Pull request by spoorthik**
k_busy_wait() call used in test expects time in us, but the test is
specifying wait in ms.

Also the test fails on NRF5 platform as the test hardcodes the interrupts
priority to 0 and 1 and assumes system timer to be of priority 0 which
is not the case in NRF5 platforms as per @pizi-nordic where system timer
is at priority 1. Hence changing test interrupts to 1 and 2.

Fixes: #10509 

Signed-off-by: Spoorthi K <spoorthi.k@intel.com>

### CODE REVIEW

**Comment by spoorthik** [comment]
recheck

File: tests/kernel/interrupt/src/nested_irq.c [hunk 1]
```diff
@@ -5,7 +5,8 @@ struct k_timer timer;
 
 #define ISR0_OFFSET 1
 #define ISR1_OFFSET 2
-
+/* ms is in milli seconds */
+#define DELAY_US(ms)  K_MSEC(ms) * USEC_PER_MSEC
```
Comment (pizi-nordic) [inline tests/kernel/interrupt/src/nested_irq.c]: The DELAY_US(ms) might be misunderstood. I assume that you want to delay XX ms and macro returns value in us.
What about MS2US(ms) or just use the `K_MSEC(ms) * USEC_PER_MSEC` as an k_busy_wait() argument?

Comment (andyross) [reply tests/kernel/interrupt/src/nested_irq.c]: Pedantry: that macro should be fully parenthesized so the multiplication binds as intended when used in an expression.  For example "delays_per_sec = 1000000 / DELAY_US()" will compute "(1000000/K_MSEC(ms)) * 1000" which is obvoiusly wrong.

File: tests/kernel/interrupt/src/nested_irq.c [hunk 2]
```diff
@@ -55,8 +56,8 @@ void isr0(void *param)
 #ifndef NO_TRIGGER_FROM_SW
 void test_nested_isr(void)
 {
-	IRQ_CONNECT(IRQ_LINE(ISR0_OFFSET), 1, isr0, NULL, 0);
-	IRQ_CONNECT(IRQ_LINE(ISR1_OFFSET), 0, isr1, NULL, 0);
+	IRQ_CONNECT(IRQ_LINE(ISR0_OFFSET), 2, isr0, NULL, 0);
+	IRQ_CONNECT(IRQ_LINE(ISR1_OFFSET), 1, isr1, NULL, 0);
```
Comment (andyross) [inline tests/kernel/interrupt/src/nested_irq.c]: FWIW: if the test is really dependent on IRQ priority being in a certain order, that deserves a comment here.  Also those priority vales are platform-dependent and probably deserve some macro hackery to make them settable per platform.

**Review by andyross — CHANGES_REQUESTED** [review]
Just barely a -1 from me for that macro hygiene issue.  Otherwise fix looks correct.

File: tests/kernel/interrupt/src/nested_irq.c (hunk 2 above)
Comment (spoorthik) [reply tests/kernel/interrupt/src/nested_irq.c]: @andyross: Thanks, I agree, provided a comment there to explain why priority is changed. The intention in the test is to get preempted by a timer while executing an ISR. So it wouldn't need a macro hackery IMHO.

File: tests/kernel/interrupt/src/nested_irq.c (hunk 1 above)
Comment (spoorthik) [reply tests/kernel/interrupt/src/nested_irq.c]: Agreed, incorporated both the reviews from @andyross and @pizi-nordic. Thanks.

**Review by pizi-nordic — APPROVED** [review]

**Review by andyross — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**